Source/render/ShapeRenderer.swift

```diff
@@ -169,6 +171,25 @@ class ShapeRenderer: NodeRenderer {
         drawGradient(gradient, ctx: ctx, opacity: opacity)
     }
 
+    fileprivate func drawPattern(_ pattern: Pattern, ctx: CGContext?, opacity: Double) {
+        guard let shape = shape else {
+            return
+        }
+        if !pattern.userSpace {
+            BoundsUtils.applyTransformToNodeInRespectiveCoords(respectiveNode: pattern.content, absoluteLocus: shape.form)
```

Comment:
This line looks like a bug for me. You're modifying pattern during rendering. We should never change user model. Moreover these changes will accumulate, so `pattern.content` should be incorrect after several draws.